### PR TITLE
stm32: Add USB support for STM32L1 MCUs.

### PR DESCRIPTION
**Pull request by yn386**
This PR supports USB feature for STM32L1 MCUs.
By default, USB is disabled because NUCLEO-L152RE does not have USB port.
To use USB feature,
* Modify definition of `MICROPY_HW_ENABLE_USB` 0 to 1 on `boards/NUCLEO_L152RE/mpconfigboard.h`
* Connect USB cable's `USB_DP` to `PA12`, `USB_DM` to `PA11`.

Tested on NUCLEO-L152RE.
* REPL with USB Virtual COM Port works.
* USB Mass Storage works (can access to internal storage).

### CODE REVIEW

File: ports/stm32/powerctrlboot.c [hunk 1]
```diff
@@ -388,25 +390,6 @@ void SystemClock_Config(void) {
     SystemCoreClockUpdate();
     powerctrl_config_systick();
 
-    #if MICROPY_HW_ENABLE_USB
```
Comment (dpgeorge) [inline ports/stm32/powerctrlboot.c:391]: Why did you remove this code?  It should work with the clock recovery system and HSI.

File: ports/stm32/usbd_conf.c [hunk 2]
```diff
@@ -75,6 +75,16 @@ void HAL_PCD_MspInit(PCD_HandleTypeDef *hpcd) {
         mp_hal_pin_config(pin_A12, MP_HAL_PIN_MODE_ANALOG, MP_HAL_PIN_PULL_NONE, 0);
         mp_hal_pin_config_speed(pin_A12, GPIO_SPEED_FREQ_VERY_HIGH);
 
+        #elif defined(STM32L1)
+
+        // STM32L1 doesn't have an alternate function for USB.
+        // To be disconnected from all peripherals, put in input mode.
+
+        mp_hal_pin_config(pin_A11, MP_HAL_PIN_MODE_INPUT, MP_HAL_PIN_PULL_NONE, 0);
```
Comment (dpgeorge) [inline ports/stm32/usbd_conf.c:83]: Is it possible to use `MP_HAL_PIN_MODE_ANALOG` here, instead of input mode?  That's how the G0 and G4 MCU's work.

File: ports/stm32/powerctrlboot.c (hunk 1 above)
Comment (yn386) [reply ports/stm32/powerctrlboot.c:391]: Because STM32L152 does not have clock recovery system and HSI48.
I noticed that `__HAL_RCC_SYSCFG_CLK_ENABLE()` is equivalent to `RCC->APB2ENR |= RCC_APB2ENR_SYSCFGEN` and it is needed only if USB is enabled.
So I modified it (a2c0201)

File: ports/stm32/usbd_conf.c (hunk 2 above)
Comment (yn386) [reply ports/stm32/usbd_conf.c:83]: I tried it before but it does not work with STM32L1. 
This is due to difference of the the port mode's reset state: G0 and G4 MCUs are analog mode, while L1 MCUs are input mode.

File: ports/stm32/powerctrlboot.c (hunk 1 above)
Comment (dpgeorge) [reply ports/stm32/powerctrlboot.c:391]: Ah, yes, you are correct, the L1 doesn't have these things.

File: ports/stm32/usbd_conf.c (hunk 2 above)
Comment (dpgeorge) [reply ports/stm32/usbd_conf.c:83]: OK, thanks for trying that.

**Comment by dpgeorge** [comment]
Thanks for the contribution!
